nsis/src/nsis/NsisLoader.java

```diff
@@ -159,6 +156,38 @@ private Data createData(Program program, Address address, DataType dt)
 		return d;
 	}
 
+	/**
+	 * 
+	 * Initializes a memory block in Ghidra with the given permissions
+	 * 
+	 * @param is                InputStream of the data
+	 * @param startingAddr
+	 * @param program
+	 * @param monitor
+	 * @param size
+	 * @param blockName
+	 * @param readPermission
+	 * @param writePermission
+	 * @param executePermission
+	 * @throws DuplicateNameException
+	 * @throws CancelledException
+	 * @throws AddressOverflowException
+	 * @throws MemoryConflictException
+	 * @throws LockException
+	 */
+	private void createGhidraMemoryBlock(InputStream is, Address startingAddr, Program program,
```

Comment:
Readability nit: group similar methods together. Right not your order of methods is: initFirstHeader, createGhidraMemoryBlock, initCommonHeader. I think it would be more logical to not split similar initFirstHeader and initCommonHeader apart.

Reply:
Resolved